bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Tree.java

```diff
@@ -2108,7 +2108,9 @@ boolean isTrim (NSView view) {
 @Override
 void keyDown(long id, long sel, long theEvent) {
 	ignoreSelect = preventSelect = false;
+	keyDown = true;
 	super.keyDown(id, sel, theEvent);
+	keyDown = false;
```

Comment:
the same is also done in Table#keyDown(). I am asking me if we should better call ```keyDown = false``` in a finally block for the case when an exception is thrown in super.keyDown. Can we ensure that no exception is thrown in super.keyDown() ?

Reply:
super.keyDown() sends the key down event and calls the keydown listeners. It's possible that there are exceptions thrown in the listener code. They have to be handled in the listener or it would be break the execution.